src/Core/Hook/HookDispatcherAwareTrait.php

```diff
@@ -36,6 +38,18 @@ trait HookDispatcherAwareTrait
      */
     protected $hookDispatcher;
 
+    /**
+     * @return HookDispatcherInterface|NullDispatcher
+     */
+    public function getHookDispatcher()
```

Comment:
I'd make this `protected` instead.

Reply:
I see your point, but it feels weir to have a public setter and a protected getter no?

Reply:
It's weird in terms that the setter shouldn't exist in the first place I guess.